tests/conftest.py

```diff
@@ -189,11 +189,13 @@ def build_env():
 
 
 class Distribution:
-    def __init__(self, key, version, editable=False):
+    def __init__(self, key, version, editable=False, metadata=None):
         self.version = version
         self.link_file = "editable" if editable else None
         self.dependencies = []
-        self.metadata = {"Name": key}
+        self.metadata = {"Name": key, "Version": version}
+        if metadata:
+            self.metadata = {**self.metadata, **metadata}
```

Comment:
Would you prefer `name` and `version` each to be a `@property` reading from `self.metadata`?